app/controllers/wallets_controller.rb

```diff
@@ -0,0 +1,42 @@
+class WalletsController < ApplicationController
+
+  def show
+    authorize @wallet
+  end
+
+  def index
+    if params[:query].present?
```

Comment:
The wallets won't have a search field, we don't think of having more than 3, 5 wallets.